obs_maven/repo.py

```diff
@@ -83,6 +94,19 @@ def parse_primary(self):
                         input_source.setByteStream(gzip_fd)
                         parser.parse(input_source)
                         self._rpms = handler.rpms.values()
+
+                        # Cache primary XML data in filesystem
+                        if not os.path.exists(self.cache_dir):
+                            logging.debug("Creating cache directory: %s", self.cache_dir)
+                            os.makedirs(self.cache_dir)
+                        else:
+                            # Delete old cache files from directory
+                            for f in os.listdir(self.cache_dir):
+                                os.remove(os.path.join(self.cache_dir, f))
+                        logging.debug("Caching RPMs in file: %s", cache_file)
+                        fw = open(cache_file, 'wb')
+                        pickle.dump(list(self._rpms), fw)
+                        fw.close()
```

Comment:
We should move writing the cache outside of the `for` loop. Otherwise potential `OSErrors` would trigger not wanted retries.
Also, the assumption that the cache directory is always writable is too optimistic in my opinion. We should add `OSErros` handling for that part.

Reply:
Thanks, done.